crates/sargon/src/profile/mfa/security_structures/roles/factor_levels/factor_instance_level/role_into_scrypto_access_rule.rs

```diff
@@ -0,0 +1,69 @@
+use crate::prelude::*;
+
+impl<const ROLE: u8> From<RoleWithFactorInstances<ROLE>> for ScryptoAccessRule {
+    fn from(value: RoleWithFactorInstances<ROLE>) -> Self {
+        let from_factors =
+            |factors: &Vec<FactorInstance>| -> Vec<ScryptoResourceOrNonFungible> {
+                factors
+                    .iter()
+                    .map(|instance| instance.badge.clone())
+                    .map(ScryptoResourceOrNonFungible::from)
+                    .collect()
+            };
+        ScryptoAccessRule::Protected(ScryptoCompositeRequirement::AnyOf(vec![
+            ScryptoCompositeRequirement::BasicRequirement(
+                ScryptoBasicRequirement::CountOf(
+                    value.get_threshold(),
+                    from_factors(value.get_threshold_factors()),
+                ),
+            ),
+            ScryptoCompositeRequirement::BasicRequirement(
+                ScryptoBasicRequirement::AnyOf(from_factors(
+                    value.get_override_factors(),
+                )),
+            ),
+        ]))
+    }
+}
+
+impl From<MatrixOfFactorInstances> for ScryptoRuleSet {
```

Comment:
This PR does in fact not make use of this `From for ScryptoRuleSet` since the ManifestBuilding method in Scrypto takes each role's AccessRule individiaully, see: https://github.com/radixdlt/sargon/pull/293/files#diff-466895b6bcee7ef733f42aa6a1077061bda36ee10b372bda4f68a3004a86f3d7R68-R74